src/fusiontrees/iterator.jl

```diff
@@ -59,105 +68,178 @@ function Base.iterate(it::FusionTreeIterator{I,0},
 end
 
 function Base.iterate(it::FusionTreeIterator{I,1},
-                      state=(it.uncoupled[1] != it.coupled)) where {I<:Sector}
+                      state=!(it.coupled ∈ it.uncouplediterators[1])) where {I<:Sector}
     state && return nothing
     T = vertex_labeltype(I)
-    tree = FusionTree{I,1,0,0,T}(it.uncoupled, it.coupled, it.isdual, (), ())
+    tree = FusionTree{I,1,0,0,T}((it.coupled,), it.coupled, it.isdual, (), ())
     return tree, true
 end
 
 #   General case:
-function Base.iterate(it::FusionTreeIterator{I,N} where {N}) where {I<:Sector}
-    next = _iterate(it.uncoupled, it.coupled)
+function Base.iterate(it::FusionTreeIterator{I}) where {I<:Sector}
+    coupled = it.coupled
+    next = _fusiontree_iterate(it.uncouplediterators, coupled)
     next === nothing && return nothing
-    lines, vertices, states = next
-    vertexlabels = labelvertices(it.uncoupled, it.coupled, lines, vertices)
-    f = FusionTree(it.uncoupled, it.coupled, it.isdual, lines, vertexlabels)
-    return f, (lines, vertices, states)
+    out, lines, vertices, states = next
+    vertexlabels = _labelvertices(out, coupled, lines, vertices)
+    f = FusionTree(out, coupled, it.isdual, lines, vertexlabels)
+    return f, (out, lines, vertices, states)
 end
-function Base.iterate(it::FusionTreeIterator{I,N} where {N}, state) where {I<:Sector}
-    next = _iterate(it.uncoupled, it.coupled, state...)
+function Base.iterate(it::FusionTreeIterator{I},
+                      (out, lines, vertices, states)) where {I<:Sector}
+    coupled = it.coupled
+    next = _fusiontree_iterate(it.uncouplediterators, coupled, out, lines, vertices, states)
     next === nothing && return nothing
-    lines, vertices, states = next
-    vertexlabels = labelvertices(it.uncoupled, it.coupled, lines, vertices)
-    f = FusionTree(it.uncoupled, it.coupled, it.isdual, lines, vertexlabels)
-    return f, (lines, vertices, states)
+    out, lines, vertices, states = next
+    vertexlabels = _labelvertices(out, it.coupled, lines, vertices)
+    f = FusionTree(out, coupled, it.isdual, lines, vertexlabels)
+    return f, (out, lines, vertices, states)
 end
 
-function labelvertices(uncoupled::NTuple{2,I}, coupled::I, lines::Tuple{},
-                       vertices::Tuple{Int}) where {I<:Sector}
+@inline function _labelvertices(uncoupled::NTuple{2,I}, coupled::I, lines::Tuple{},
```

Comment:
That's probably a good idea. I don't know why I thought that people would ever come up with custom names for this.

Reply:
Looking at removing this, it seems like this would never have worked with anything other than integers. The values in `f.vertices` are used to index into `Fsymbol` and `Rsymbol`, so they need to be integers and not labels :-)

Reply:
@lkdvos , can you look at removing any reference to vertex labels in TensorKitSectors.jl?